src/libsyntax/parse/parser/item.rs

```diff
@@ -295,12 +305,19 @@ impl<'a> Parser<'a> {
             } else {
                 Abi::Rust
             };
+
+            let constness = if self.eat_keyword(kw::Const) {
```

Comment:
I think this is the wrong grammar. It should be `const unsafe extern "C" ...` and `const extern "C" ...` because the grammar is `const unsafe` not `unsafe const`.